ddl/partition.go

```diff
@@ -2446,6 +2453,21 @@ func (w *worker) onExchangeTablePartition(d *ddlCtx, t *meta.Meta, job *model.Jo
 			job.State = model.JobStateRollingback
 			return ver, errors.Trace(err)
 		}
+		if variable.EnableCheckConstraint.Load() {
```

Comment:
I would prefer if the check is combined with the `checkExchangePartitionRecordValidation` above, to only do a single scan of the non-partitioned table.